identity/resource_group.go

```diff
@@ -124,49 +101,38 @@ func ResourceGroup() *schema.Resource {
 				ForceNew: true,
 				Required: true,
 			},
-			"allow_cluster_create": {
-				Type:     schema.TypeBool,
-				Optional: true,
-			},
-			"allow_sql_analytics_access": {
-				Type:     schema.TypeBool,
-				Optional: true,
-			},
-			"allow_instance_pool_create": {
-				Type:     schema.TypeBool,
-				Optional: true,
-			},
 			"url": {
 				Type:     schema.TypeString,
 				Computed: true,
 			},
 		},
 	}
-}
-
-func isGroupClusterCreateEntitled(group *ScimGroup) bool {
-	for _, entitlement := range group.Entitlements {
-		if entitlement.Value == AllowClusterCreateEntitlement {
-			return true
+	//Add entitlements to the schema
+	for _, entitlementField := range entitlementFields {
+		groupResource.Schema[entitlementField] = &schema.Schema{
+			Type:     schema.TypeBool,
+			Optional: true,
 		}
 	}
-	return false
+	return groupResource
 }
 
-func isGroupSQLAnalyticsAccessEntitled(group *ScimGroup) bool {
-	for _, entitlement := range group.Entitlements {
-		if entitlement.Value == AllowSQLAnalyticsAccessEntitlement {
+func groupEntitlementExists(group *ScimGroup, entitlement Entitlement) bool {
```

Comment:
make `type Entitlements []entitlementsListItem` and add make this as a method of that type, probably call it as `Has(fieldName string) bool`. you can replace the underscore in the field name to match it with an entitlement.